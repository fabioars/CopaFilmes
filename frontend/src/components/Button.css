.Button {
    background: var(--primary-color);
    color: #fff;
    font-weight: bold;
    text-transform: uppercase;
    border: 0;
    padding: 0 44px;
    min-height: 55px;
    cursor: pointer;
    transition: all ease 300ms;
}

.Button:hover {
    background-color: #222;
}

.Button:disabled {
    opacity: 0.5;
    pointer-events: none;
    transform: scale(0.95);
}
