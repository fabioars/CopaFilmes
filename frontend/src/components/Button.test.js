import React from 'react';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Button from './Button';

describe('Button', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        render(<Button>Hello</Button>, div);
        unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Button>Hello</Button>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly disabled', () => {
        const tree = renderer.create(<Button disabled>Hello</Button>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should call the function on click', () => {
        const callback = jest.fn();
        const div = document.createElement('div');
        document.body.appendChild(div);

        act(() => {
            render(<Button onClick={callback}>Hello</Button>, div);
        });

        div.querySelector('button').click();

        expect(callback.mock.calls.length).toBe(1);
        unmountComponentAtNode(div);
    })
});
