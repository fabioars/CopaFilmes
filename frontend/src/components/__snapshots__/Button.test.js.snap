// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="Button"
>
  Hello
</button>
`;

exports[`Button renders correctly disabled 1`] = `
<button
  className="Button"
  disabled={true}
>
  Hello
</button>
`;
